src/engine/Settings.cs

```diff
@@ -179,59 +285,55 @@ public void ApplySoundLevels()
         AudioServer.SetBusMute(music, VolumeMusicMuted);
     }
 
+    /// <summary>
+    ///   Applies current window settings to any applicable engine systems.
+    /// </summary>
     public void ApplyWindowSettings()
     {
         OS.WindowFullscreen = FullScreen;
         OS.VsyncEnabled = VSync;
     }
 
-    public void ApplyGraphicsSettings()
-    {
-        GUICommon.Instance.GetTree().Root.GetViewport().Msaa = MSAAResolution;
-        ColourblindScreenFilter.Instance.SetColourblindSetting(ColourblindSetting);
-    }
-
-    /// <summary>
-    ///   Reset all options to default values
-    /// </summary>
-    public void ResetToDefaults()
-    {
-        var defaults = new Settings();
-        var type = GetType();
-
-        foreach (var property in type.GetProperties())
-        {
-            if (!property.CanWrite)
-                continue;
-
-            property.SetValue(this, property.GetValue(defaults));
-        }
-    }
-
+    // Creates and returns a settings object loaded from the configuration settings file, or defaults if that fails.
     private static Settings LoadSettings()
-    {
-        var settings = LoadSettingsFileOrDefault();
-        settings.ApplyAll();
-        return settings;
-    }
-
-    private static Settings LoadSettingsFileOrDefault()
     {
         using (var file = new File())
         {
+            Settings settings;
+
             var error = file.Open(Constants.CONFIGURATION_FILE, File.ModeFlags.Read);
 
             if (error != Error.Ok)
             {
-                GD.Print("Settings file is missing or unreadable, using default settings");
-                return new Settings();
+                GD.Print("Settings configuration file is missing or unreadable, using default settings.");
+
+                settings = new Settings();
+                settings.Save();
+
+                return settings;
             }
 
             var text = file.GetAsText();
 
             file.Close();
 
-            return JsonConvert.DeserializeObject<Settings>(text);
+            settings = JsonConvert.DeserializeObject<Settings>(text);
+
+            return settings;
+        }
+    }
+
+    // Copies all properties from another settings object to the current one.
```

Comment:
This should be an xml comment.

Reply:
I had avoided xml comments for private methods because I've seen very few of them with comments at all, and the style guide specified them for public and protected methods. If I should start doing it for private too then that's fine. I just wasn't aware.

Reply:
There isn't a requirement to document private methods, as I didn't enable requiring all elements to be documented. And also I adopted the lazy-ish style of not commenting private methods. But when there is a comment on them it should be an xml comment.